cadc-uws-server/src/main/java/ca/nrc/cadc/uws/util/RestletWebUtil.java

```diff
@@ -0,0 +1,94 @@
+ *  written permission.                  préalable et particulière
+ *                                       par écrit.
+ *
+ *  This file is part of the             Ce fichier fait partie du projet
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.uws.util;
+
+import org.restlet.Request;
+import org.restlet.data.Form;
+import org.restlet.data.Parameter;
+import ca.nrc.cadc.net.NetUtil;
+
+public class RestletWebUtil {
+
+    /**
+     * Restlet request constructor that taken a path override parameters.
+     * <p>
+     * TODO: This method REQUIRES Restlet 2.0.2.
+     * TODO: jenkinsd 2018.02.06
+     *
+     * @param request The Request object.
+     */
+    public static String getClientIP(final Request request) {
+        final Form requestHeaders = (Form) request.getAttributes().get("org.restlet.http.headers");
+        final Parameter forwardedClientIP = (requestHeaders == null)
+            ? null : requestHeaders.getFirst(NetUtil.FORWARDED_FOR_CLIENT_IP_HEADER);
+        return (forwardedClientIP == null)
+            ? request.getClientInfo().getAddress() : forwardedClientIP.getValue().split(",")[0];
+    }
+}
```

Comment:
Too bad we couldn't use NetUtil here but I guess this is the best we can do.  Your note about Restlet 2.0.2--is that what we're running everywhere?

Reply:
I take back the comment.  I misread the API as I thought `getAttributes()` was deprecated after that version.